src/test/lib/utils/meta_tables/meta_exec_table_test.cpp

```diff
@@ -0,0 +1,115 @@
+#include "base_test.hpp"
+
+#include "operators/table_wrapper.hpp"
+#include "utils/meta_tables/meta_exec_table.hpp"
+
+#include "../plugin_test_utils.hpp"
+
+namespace opossum {
+
+class MetaExecTest : public BaseTest {};
+
+TEST_F(MetaExecTest, IsMutable) {
+  const auto meta_exec_table = std::make_shared<MetaExecTable>();
+
+  EXPECT_TRUE(meta_exec_table->can_insert());
+  EXPECT_FALSE(meta_exec_table->can_update());
+  EXPECT_FALSE(meta_exec_table->can_delete());
+}
+
+TEST_F(MetaExecTest, SelectUserExecutableFunctions) {
+  auto& pm = Hyrise::get().plugin_manager;
+
+  pm.load_plugin(build_dylib_path("libhyriseTestPlugin"));
+  pm.load_plugin(build_dylib_path("libhyriseSecondTestPlugin"));
+
+  const auto expected_table = std::make_shared<Table>(
+      TableColumnDefinitions{{"plugin_name", DataType::String, false}, {"function_name", DataType::String, false}},
+      TableType::Data, 5);
+
+  expected_table->append({pmr_string{"hyriseSecondTestPlugin"}, pmr_string{"OurFreelyChoosableFunctionName"}});
+  expected_table->append({pmr_string{"hyriseTestPlugin"}, pmr_string{"OurFreelyChoosableFunctionName"}});
+  expected_table->append({pmr_string{"hyriseTestPlugin"}, pmr_string{"SpecialFunction17"}});
+
+  auto table_wrapper = std::make_shared<TableWrapper>(std::move(expected_table));
+  table_wrapper->execute();
+
+  const auto& [_, result_table] = SQLPipelineBuilder{"SELECT * FROM meta_exec"}.create_pipeline().get_result_table();
+  EXPECT_TABLE_EQ_UNORDERED(result_table, table_wrapper->get_output());
+}
+
+TEST_F(MetaExecTest, CallUserExecutableFunctions) {
+  auto& pm = Hyrise::get().plugin_manager;
+  auto& sm = Hyrise::get().storage_manager;
+
+  pm.load_plugin(build_dylib_path("libhyriseTestPlugin"));
+  pm.load_plugin(build_dylib_path("libhyriseSecondTestPlugin"));
+
+  SQLPipelineBuilder{
+      "INSERT INTO meta_exec (plugin_name, function_name) VALUES ('hyriseTestPlugin', "
+      "'OurFreelyChoosableFunctionName')"}
+      .create_pipeline()
+      .get_result_table();
+  // The test plugin creates the below table when the called function is executed
+  EXPECT_TRUE(sm.has_table("TableOfTestPlugin"));
+
+  SQLPipelineBuilder{
+      "INSERT INTO meta_exec (plugin_name, function_name) VALUES ('hyriseSecondTestPlugin', "
+      "'OurFreelyChoosableFunctionName')"}
+      .create_pipeline()
+      .get_result_table();
+  // The second test plugin creates the below table when the called function is executed
+  EXPECT_TRUE(sm.has_table("TableOfSecondTestPlugin"));
+}
```

Comment:
These and the following tests kind of mirror some of the PluginManagerTests. It's pretty much the except that the SQLPipeline and everything that comes with it is in between. However, it's still nice to see everything working from end to end but we don't have to keep it.